operator/builtin/input/file/config.go

```diff
@@ -90,7 +90,9 @@ func (c InputConfig) Build(context operator.BuildContext) ([]operator.Operator,
 		return nil, fmt.Errorf("`max_log_size` must be positive")
 	}
 
-	if c.MaxConcurrentFiles <= 1 {
+	if c.MaxConcurrentFiles == 0 {
+		c.MaxConcurrentFiles = defaultMaxConcurrentFiles
```

Comment:
I think `NewInputConfig` already sets this to 1024. I added print statements to a build (from master) and verified that default is 1024, and if I set it to 10 (or any other value), it is respected.

In fact, with 40k files in a directory, setting to 10 prevents "too many open files". I think we might want to choose something less than 1024, maybe 1000 (this works on my test instance).

Reply:
Ah, good point. I'll close this PR.

Regarding the default though - you're saying 1000 works but 1024 does not? Seems like an unsubstantial change. Does 1000 come from anywhere in particular?

Reply:
On most systems, 1024 will be the default limit for a process (I believe). Stanza already has a file handle open to a couple things by default. When file_input demands 1024, we hit the limit earlier than expected.

Reply:
Makes sense. Should we give it more margin than 24 then? Maybe set it to 512?

Reply:
I think 512 is more than reasonable. Anything higher, the user should really be defining their upper limit.